src/spikeinterface/core/base.py

```diff
@@ -846,43 +846,63 @@ def save_to_memory(self, sharedmem=True, **save_kwargs) -> "BaseExtractor":
         return cached
 
     # TODO rename to saveto_binary_folder
-    def save_to_folder(self, name=None, folder=None, overwrite=False, verbose=True, **save_kwargs):
+    def save_to_folder(
+        self,
+        name: str = None,
```

Comment:
```suggestion
        name: str | Path | None = None,
```

Or you don't like adding None as a potential type hint? And below you say Path so it needs Path at minimum.

Reply:
I think that's good. But I don't think name can take a Path. Am I wrong?

Side comment, talking about pet peeves this is one of mine. I actually prefer Optional[str] instead of str | None.  But  I think this is inconsequential and I have been following your standard in this one.

Reply:
So my comment was originally because the docstring below said str or Path. But looking at the code it has to be a path because of this line

```python
folder = cache_folder / name
```
`/` can't be used with string. Or we need to do name = Path(name) to make sure it is a path first.

Reply:
> Side comment, talking about pet peeves this is one of mine. I actually prefer Optional[str] instead of str | None. But I think this is inconsequential and I have been following your standard in this one.

I'm okay with Optional[str], but I don't like when someone does Optional[list[float|int']] because it gets too nested. I don't like the numpy default of doing argument, optional to indicate that a default argument is supplied.

Reply:
> / can't be used with string. Or we need to do name = Path(name) to make sure it is a path first.

You totally can:

```python
In [1]: from pathlib import Path

In [2]: Path.cwd() / "here"
Out[2]: PosixPath('/home/heberto/development/neuroconv/here')
```

Concerning the typing discussion, this is just a small preference. I would rather keep doing it consistently with what you have been doing : )

Reply:
Sorry you're right. I wasn't thinking that folder could be a string. But you can also do:
```
first_path = Path('.')
second_path = Path('folder')
full_path = first_path / second_path
```

So it can be a string or Path.

Reply:
Oh fair enough. 
I guess that you are right that it **can** be a path and that would work. I don't think this is what it was intended and I would rather not give a type hint for this.

Reply:
Then I would delete it from the docstring as well.

Reply:
Oh, yes, sure! 

I think that the intention is for you to pass the name of the file. As you mentioned you could pass Path("name") and that would work. But the reason that I think is a bad idea is because if someone is relying on a linter and then they pass Path("C:/something_here/here_something_else/") then that would throw an error but the linter would accept it. 

To be honest, my intuition is that folder is sufficient as it can do the two things. But I think that probably at some point it was found convenient to save to the tmp folder but still name it.

Reply:
Should be done.